nemo-physical/src/tabular/operations/materialize.rs

```diff
@@ -72,17 +75,29 @@ pub fn materialize(trie_scan: &mut impl TrieScan) -> Option<Trie> {
     materialize_up_to(trie_scan, 0)
 }
 
-/// Tests whether an iterator is empty by materializing it until the first element
-pub fn scan_is_empty(trie_scan: &mut impl TrieScan) -> bool {
-    trie_scan
-        .advance_on_layer(trie_scan.column_types().len())
-        .is_some()
+/// Tests whether an iterator is empty by materializing it until the first element.
+/// Returns the first row of the result or `None` if it is empty.
+pub fn scan_is_empty(trie_scan: &mut impl TrieScan) -> Option<TableRow> {
```

Comment:
Since this no longer returns boolean, the name should change to reflect that, maybe `scan_first_match`?